zerver/tests/test_messages.py

```diff
@@ -2092,3 +2095,280 @@ def test_delete_message_second_time(self):
         self.client_delete('/json/messages/{msg_id}'.format(msg_id=msg_id))
         result = self.client_delete('/json/messages/{msg_id}'.format(msg_id=msg_id))
         self.assert_json_error(result, "Invalid message(s)")
+
+class SoftDeactivationMessageTest(ZulipTestCase):
+
+    def soft_deactivate_user(self, user):
+        # type: (str) -> UserProfile
+        long_term_idle_user = UserProfile.objects.get(
+            email=self.example_email(user))
+        long_term_idle_user.last_active_message_id = UserMessage.objects.filter(
+            user_profile=long_term_idle_user).order_by(
+            '-message__id')[0].message_id
+        long_term_idle_user.long_term_idle = True
+        long_term_idle_user.save(update_fields=[
+            'long_term_idle',
+            'last_active_message_id'])
+        return long_term_idle_user
+
+    def get_user_ums(self, user_profile):
+        # type: (UserProfile) -> List[UserMessage]
+        return list(UserMessage.objects.filter(
+            user_profile=user_profile).order_by('message__id'))
+
+    def test_maybe_catch_up_soft_deactivated_user(self):
+        # type: () -> None
+        recipient_list  = [self.example_email("hamlet"), self.example_email("iago")]
+        for email in recipient_list:
+            self.subscribe_to_stream(email, "Denmark")
+
+        sender = self.example_user('iago')
+        realm = sender.realm
+        sending_client = make_client(name="test suite")
+        stream = get_stream('Denmark', realm)
+        subject = 'foo'
+
+        def last_realm_audit_log_entry(event_type):
+            # type: (str) -> RealmAuditLog
+            return RealmAuditLog.objects.filter(
+                event_type=event_type
+            ).order_by('-event_time')[0]
+
+        def send_fake_message(message_content):
+            # type: (str) -> Message
+            recipient = get_recipient(Recipient.STREAM, stream.id)
+            return Message.objects.create(sender = sender,
+                                          recipient = recipient,
+                                          subject = subject,
+                                          content = message_content,
+                                          pub_date = timezone_now(),
+                                          sending_client = sending_client)
```

Comment:
Actually my way of doing things was what required this initially. There on as I build the feature things got mutated and I didn't thought of doing a re arrangement of commits that would allow sending an actual message here.
Basically during the time of review, sending a actual message would have created a UserMessage row for long_term_idle user anyway.
Now to address this I have refactored commits and rearranged them to allow us send an actual message here :)